index.bs

```diff
@@ -485,6 +501,159 @@ spec:csp3; type:grammar; text:base64-value
+
+  The headers' value is a <a>Dictionary</a> [[RFC9651]], with every member-value being an
+  <a>inner list</a> of <a>tokens</a>.
+  
+  A <dfn>source</dfn> is a string. The only possible value for it is "`inline`".
+
+  A <dfn>destination</dfn> is a <a for=request>destination</a>. The only possible value for it is "`script`".
+
+  An <dfn>integrity policy struct</dfn>, is a <a>struct</a> that contains the following:
+
+    * <dfn>sources</dfn>, a list of <a>source</a>s, initially empty.
+    * <dfn>blocked destinations</dfn>, a list of <a>destination</a>s, initially empty.
+    * <dfn>endpoints</dfn>, a <a for=/>list</a> of strings, initially empty.
+
+  When <dfn>processing an integrity policy</dfn>, with |headers| and |header name|, do the following:
+
+  1. Let |integrity policy| be a new <a>integrity policy struct</a>.
+  1. Let |dictionary| be the result of <a>getting a structured field value</a> from |headers|
+     given |header name| and "`dictionary`".
+  1. If |dictionary|["`sources`"] does not <a for=list>exist</a> or if its value
+     <a for=list>contains</a> "`inline`", <a for=list>append</a> "`inline`" to
+     |integrity policy|'s <a>sources</a>.
+  1. If |dictionary|["`blocked-destinations`"] <a for=list>exists</a>:
+    1. If its value <a for=list>contains</a> "`script`",
+       <a for=list>append</a> "`script`" to |integrity policy|'s <a>blocked destinations</a>.
+  1. If |dictionary|["`endpoints`"] <a for=map>exists</a>:
+    1. Set |integrity policy|'s <a>endpoints</a> to |dictionary|['endpoints'].
+  1. Return |integrity policy|.
+  
+  ### Parse Integrity-Policy headers ### {#parse-integrity-policy-headers-section}
+  To <dfn>parse Integrity-Policy headers</dfn>, given a <a for=/>Response</a> |response|
+  and a <a for=/>policy container</a> |container|, do the following:
+
+  1. Let |headers| be |response|'s <a for=response>header list</a>.
+  1. If |headers| <a for="header list">contains</a> a <a>header</a>
+     whose <a>header name</a> is "`Integrity-Policy`",
+     set |container|'s <a>integrity policy</a> be the result of running
+     <a>processing an integrity policy</a> with the corresponding <a>header value</a>.
+  1. If |headers| <a for="header list">contains</a> a <a>header</a>
+     whose <a>header name</a> is "`Integrity-Policy-Report-Only`",
+     set |container|'s <a>report only integrity policy</a> be the result of running
+     <a>processing an integrity policy</a> with the corresponding <a>header value</a>.
+  
+  ### Should request be blocked by Integrity Policy ### {#should-request-be-blocked-by-integrity-policy-section}
+  To determine <dfn>should request be blocked by integrity policy</dfn>, given a <a for=/>request</a> |request|,
+  do the following:
+
+  1. Let |policy container| be |request|'s <a for=request>policy container</a>.
+  1. If |request|'s <a>integrity metadata</a> is not the empty string
+     and |request|'s <a for=request>mode</a> is either "cors" or "same-origin",
```

Comment:
These should be marked up as strings, e.g., "`cors`".

Reply:
I had to do a double-take on this. I think you want to say it should be "\`cors\`", right?

Reply:
Done